geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHScanIntegrationTest.java

```diff
@@ -219,13 +298,13 @@ public void givenCount_returnsAllEntriesWithoutDuplicates() {
       cursor = result.getCursor();
     } while (!result.isCompleteIteration());
 
-    assertThat(allEntries).hasSize(3);
-    assertThat(new HashSet<>(allEntries)).isEqualTo(entryMap.entrySet());
+    assertThat(allEntries.size()).isCloseTo(3, Offset.offset(1));
```

Comment:
see comment at the top of the test - "// note: As per redis docs, count parameter is not guaranteed to be honored."  so it's weird one to test...

Reply:
A few thoughts:
- The test name is kinda misleading, since we say `returnsAllEntriesWithoutDuplicates`, but we are admitting the possibility of duplicates.
- You could use lambdas (like you did in another part of the code) to remove any duplicates before the assertion, something like:
```
List<Map.Entry<String, String>> allEntriesWithoutDuplicates = allEntries.stream().distinct().collect(Collectors.toList());
```
- But really none of the above matters because, thinking about it more, I realize that for most of our tests, the data is so small that native Redis is always going to return the entire structure at once (per the `Why SCAN may return all the items of an aggregate data type in a single call?` section of the SCAN documentation as well as the Redis code (see `db.c` file, `scanGenericCommand` method)). So we can actually check that the size and entries match exactly what we expect.